src/monexamine.cpp

```diff
@@ -313,6 +321,60 @@ bool monexamine::mech_hack( monster &z )
     return false;
 }
 
+static int prompt_for_amount( const char *const msg, const int max )
+{
+    const std::string formatted = string_format( msg, max );
+    const int amount = string_input_popup()
+                       .title( formatted )
+                       .width( 20 )
+                       .text( to_string( max ) )
+                       .only_digits( true )
+                       .query_int();
+
+    return ( amount > max ) ? max : ( amount <= 0 ) ? 0 : amount;
```

Comment:
I don't know whether it's more readable, but here's more compact version without double ternaries.
```suggestion
    return std::min( max, std::max( 0, amount ) );
```

Reply:
There's also `clamp( amount, 0, max )`.

Reply:
Live and learn.

Reply:
About that part of the code,  it's copy/pasted from iexamine.cpp line 411. So should I make it into a function instead? If yes in what namespace?
If no should I also apply AMurkin's suggestion in the iexamine version?